dockerenforcer/killer.py

```diff
@@ -96,6 +104,19 @@ def should_be_killed(self, container):
                 logger.error("During execution of rule {0} exception was raised: {1}".format(rule['name'], e))
         return Verdict(False, container, None)
 
+    def all_violated_rules(self, container):
+        if not container:
+            return []
+
+        v_rules = []
+        for rule in self.__rules:
+            try:
+                if rule['rule'](container):
+                    v_rules.append(rule['name'])
```

Comment:
IMO, we need to catch violations as they occur during real checks, and not just re-execute everything here to check what was broken. This is a huge CPU waste